mne/forward/_compute_forward.py

```diff
@@ -180,63 +195,129 @@ def _make_ctf_comp_coils(info, coils):
+        coil['rmag']; position vector of coil (coil integration points?)
+    cosmag : N x 3 array
+        N 3-space vectors
+        coil['cosmag']; direction of the coil (coil integration points?)
+
+    Returns
+    -------
+    """
+    # rr, rmag refactored according to Equation (19) in Mosher, 1999
     # Knowing that we're doing all directions, the above can be refactored:
-    diff = rp.T[np.newaxis, :, :] - rr[:, :, np.newaxis]
+
+    # diff = (n_sources x 3 x n_coils pts)
+    diff = rmag.T[np.newaxis, :, :] - rr[:, :, np.newaxis]
+    # Get magnitude of vector (distance) between sensor and source cubed
+    # diff_norm = (n_sources x n_coils pts)
     diff_norm = np.sum(diff * diff, axis=1)
     diff_norm *= np.sqrt(diff_norm)
     diff_norm[diff_norm == 0] = 1  # avoid nans
+
     # This is the result of cross-prod calcs with basis vectors,
-    # as if we had taken (Q=np.eye(3)), then multiplied by the cosmags (c)
+    # as if we had taken (Q=np.eye(3)), then multiplied by cosmags
     # factor, and then summed across directions
-    x = np.array([diff[:, 1] * c[:, 2] - diff[:, 2] * c[:, 1],
-                  diff[:, 2] * c[:, 0] - diff[:, 0] * c[:, 2],
-                  diff[:, 0] * c[:, 1] - diff[:, 1] * c[:, 0]])
-    return np.rollaxis(x / diff_norm, 1)
+    # x = diff x cosmag (diff cross c); x.shape = (3 x n_sources x n_coils pts)
+    x = np.array([diff[:, 1] * cosmag[:, 2] - diff[:, 2] * cosmag[:, 1],
+                  diff[:, 2] * cosmag[:, 0] - diff[:, 0] * cosmag[:, 2],
+                  diff[:, 0] * cosmag[:, 1] - diff[:, 1] * cosmag[:, 0]])
+    return np.rollaxis(x / diff_norm, 1)  # return (n_sources x 3 x n_coil pts)
 
 
-def _bem_pot_or_field(rr, mri_rr, mri_Q, coils, solution, srr,
+def _bem_pot_or_field(rr, mri_rr, mri_Q, coils, solution, bem_rr,
                       n_jobs, coil_type):
     """Calculate the magnetic field or electric potential
 
     The code is very similar between EEG and MEG potentials, so we'll
-    combine them.
-
-    This does the work of "fwd_comp_field" (which wraps to "fwd_bem_field") and
-    "fwd_bem_pot_els" in MNE-C.
+    combine them. This does the work of "fwd_comp_field"
+    (which wraps to "fwd_bem_field") and "fwd_bem_pot_els" in MNE-C.
+
+    Parameters
+    ----------
+    rr : ndarray, shape (n_dipoles, 3)
+        Source positions in head coordinates
+    mri_rr : ndarray, shape (n_dipoles, 3)
+        Source positions in MRI coordinates
+    mri_Q :
+        3x3 head -> MRI transform. I.e., head_mri_t.dot(np.eye(3))
+    coils : list
+        List of coils where each element contains coil specific information
+    solution : ndarray, shape (n_coils, n_bem_rr)
+        Comes from _bem_specify_coils
+    bem_rr : ndarray, shape (n_bem_rr, 3)
+        BEM source locations concatenated from all layers in MRI space.
+    n_jobs : int
+        Number of jobs to run in parallel
+    coil_type : str
+        'meg' or 'eeg'
+
+    Returns
+    -------
+    B
```

Comment:
This is the forward solution IIRC. Check the shape, should be (n_src * 3) x n_sensors or the reverse